osp/core/utils/general.py

```diff
@@ -214,165 +295,255 @@ def get_custom_datatype_triples():
-    return requests.post(url=url,
-                         data=serialized,
-                         headers={"content_type": "application/json"})
+    cuds_object.add(*args, rel=rel)
+    return cuds_object
 
 
-def serialize(cuds_object, rel=cuba.activeRelationship,
-              max_depth=float("inf"), json_dumps=True):
-    """Serialize a cuds objects and all of its contents recursively.
+def get_relationships_between(subj, obj):
+    """Get the set of relationships between two cuds objects.
 
     Args:
-        cuds_object (Cuds): The cuds object to serialize
-        rel (OntologyRelationship, optional): The relationships to follow when
-            serializing recursively. Defaults to cuba.activeRelationship.
-        max_depth (int, optional): The maximum recursion depth.
-            Defaults to float("inf").
-        json_dumps (bool, optional): Whether to dump it to the registry.
-            Defaults to True.
+        subj (Cuds): The subject
+        obj (Cuds): The object
 
     Returns:
-        Union[str, List]: The serialized cuds object.
+        Set[OntologyRelationship]: The set of relationships between subject
+            and object.
     """
-    from osp.core.session.transport.transport_utils import serializable
-    from osp.core.utils import find_cuds_object
-    cuds_objects = find_cuds_object(criterion=lambda _: True,
-                                    root=cuds_object,
+    result = set()
+    for rel, obj_uids in subj._neighbors.items():
+        if obj.uid in obj_uids:
+            result.add(rel)
+    return result
+
+
+def delete_cuds_object_recursively(cuds_object, rel=cuba.activeRelationship,
+                                   max_depth=float("inf")):
+    """Delete a cuds object  and all the object inside of the container of it.
+
+    Args:
+        cuds_object (Cuds): The CUDS object to recursively delete.
+        rel (OntologyRelationship, optional): The relationship used for
+            traversal. Defaults to cuba.activeRelationship.
+        max_depth (int, optional):The maximum depth of the recursion.
+            Defaults to float("inf"). Defaults to float("inf").
+    """
+    from osp.core.utils.simple_search import find_cuds_object
+    cuds_objects = find_cuds_object(criterion=lambda x: True, root=cuds_object,
                                     rel=rel,
                                     find_all=True,
                                     max_depth=max_depth)
-    result = serializable(cuds_objects, partition_cuds=False, mark_first=True)
-    if json_dumps:
-        return json.dumps(result)
-    return result
+    for obj in cuds_objects:
+        obj.session.delete_cuds_object(obj)
 
 
-def deserialize(json_doc, session=None, buffer_context=None):
-    """Deserialize the given json objects (to CUDS).
+def import_(path: Optional[str] = None,
+            file: Union[str, dict, TextIO] = None,
+            session: Optional = None,
```

Comment:
Make it compatible with `with` statement

Reply:
Done